src/main/webapp/app/upload/upload.component.ts

```diff
@@ -12,12 +19,84 @@ export class UploadComponent {
     error: string;
     uploadResponse = { status: '', message: '' };
 
+    currentPool: Map<String, Pool> = new Map<String, Pool>();
+    newPool: Map<String, Pool> = new Map<String, Pool>();
+
+    public poolType = PoolType;
+    public isCollapsed = true;
     public files: UploadFile[] = [];
-    file: File;
 
-    constructor(private uploadService: UploadService, private router: Router) {}
+    public currentConflictPool: Pool[] = [];
+    public newConflictPool: Pool[] = [];
+    public merged: Pool[] = [];
+    public choosePattern = new Map();
+    public file: File;
+
+    constructor(private uploadService: UploadService, private router: Router, private poolService: PoolListService) {}
+
+    async onSubmit() {
+        if (!this.file) {
+            return (this.error = 'Please insert file');
+        }
+        this.clearField();
+        const formData = new FormData();
+        formData.append('file', this.file);
+
+        const oldOnesPromise = this.poolService.getPools().toPromise();
+        const newOnesPromise = this.uploadService.parse(formData).toPromise();
+
+        const oldOnes = await oldOnesPromise;
+        const newOnes = await newOnesPromise;
+
+        oldOnes.body.forEach(pool => this.currentPool.set(pool.poolId, pool));
+        newOnes.body.forEach(pool => this.newPool.set(pool.poolId, pool));
+
+        if (this.currentConflictPool.length === 0 && this.newConflictPool.length === 0) {
+            this.getConflicts();
+        }
+        this.isCollapsed = false;
```

Comment:
I think you should add another if like there here `if (this.currentConflictPool.length !== 0 && this.newConflictPool.length !== 0)`. I know it's a corner case when user tries to upload same file twice, but I encountered it. It shows empty conflict table. Or we can just ignore this fact because it will require more logic after if, up to you.